re.c

```diff
@@ -3323,7 +3323,9 @@ rb_reg_match_m(int argc, VALUE *argv, VALUE re)
 	pos = 0;
     }
 
-    str = SYMBOL_P(str) ? rb_sym2str(str) : StringValue(str);
+    if (NIL_P(str)) {
+        rb_warn("given argument is nil");
```

Comment:
I think it could be clearer that this behaviour is deprecated and will become an error:

```suggestion
        rb_warn("given argument is nil; this will raise a TypeError in the next release");
```

Reply:
Thank you! I used this change in https://github.com/ruby/ruby/pull/2637/commits/bb7f955fd1bcc3df1cefac5b530eb8aa86235edc

Reply:
Changes spec in https://github.com/ruby/ruby/pull/2637/commits/d20d9cedb9e37c03aa4d93c535b002fc6d5303d4